inductiva/utils/optional_deps.py

```diff
@@ -0,0 +1,66 @@
+"""Utils to check if optional dependencies are installed.
+
+Example usage, e.g., to check if numpy is installed:
+
+def _check_numpy():
+    import numpy
+
+extra_msg = "You can install it with `pip install numpy`."
+
+needs_numpy = functools.partial(
+    _needs_optional_deps,
+    _check_numpy,
+    extra_msg,
+)
+
+# Now use `needs_numpy` as a decorator for functions that need numpy.
+@optional_deps.needs_numpy
+def load_from_np():
+    ...
+"""
+
+import functools
+
+
+def _needs_optional_deps(check_fn, extra_info, func):
+    """Decorator to check if optional dependencies are installed."""
+
+    def wrapper(*args, **kwargs):
+        try:
+            check_fn()
+        except ImportError as e:
+            msg = (f"{str(e)}, required to use function "
+                   f"'{func.__qualname__}'.")
+
+            if extra_info:
+                msg += f" {extra_info}"
+
+            raise RuntimeError(msg) from e
+
+        return func(*args, **kwargs)
+
+    return wrapper
+
+
+def _check_molecules_optional_deps():
+    # pylint: disable=import-outside-toplevel
+    import MDAnalysis as mda
```

Comment:
Can you explain why we need to import inside and then del? I am confused by this?

Reply:
i rewrote this in a different way. Can you check if you think it's understandable?

Reply:
Much more! Thanks!